core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -154,6 +180,44 @@ package object internal {
   /** Marker trait for modules that are not true modules */
   private[chisel3] trait PseudoModule extends BaseModule
 
+  /** Computes a name
+    *
+    * @param prefix The prefix associated with the seed
+    * @param seed The seed for computing the name (if available)
+    * @param defaultPrefix Optional default prefix which is only used if the seed is not defined
+    * @param defaultSeed Optional default (or "fallback") seed
+    * @return the name, if it can be computed
+    */
+  private[chisel3] def computeName(
+    prefix:        Prefix,
+    seed:          Option[String],
+    defaultPrefix: Option[String],
+    defaultSeed:   Option[String]
+  ): Option[String] = {
+
+    /** Computes a name of this signal, given the seed and prefix */
+    def buildName(seed: String, prefix: Prefix): String = {
+      val builder = new StringBuilder()
+      prefix.foreach { p =>
+        builder ++= p
+        builder += '_'
+      }
+      builder ++= seed
+      builder.toString
+    }
+
+    seed match {
+      case Some(s) => Some(buildName(s, prefix.reverse))
+      case None =>
+        defaultSeed.map { default =>
+          defaultPrefix match {
+            case Some(p) => buildName(default, p :: prefix.reverse)
```

Comment:
shouldn't this be defaultPrefix.reverse?

Reply:
Yeah, for the record since I'm just factoring this method out, I wouldn't worry too much about the implementation as is. We should refactor it but that's not the purpose of this PR.

That being said, I believe `defaultPrefix` is only every used to insert `_` at the beginning of temporaries, I think we can capture that functionality more directly in a follow on PR.